### ISSUE TITLE
no-unreachable-types: Interface implementing other interfaces are always unreachable

### ISSUE THREAD

**Issue by tobiasdiez** [dotansimha/graphql-eslint#571]
**Describe the bug**

If you have one interface SUB that implements another interface PARENT, then the interface SUB and all types implementing this interface are marked as unreachable - although PARENT is used and is correctly not marked as unreachable.

**To Reproduce**
```graphql
interface User {
  id: ID!
}

interface Manager implements User {
  id: ID!
}

type TopManager implements Manager {
  id: ID!
  name: String
}

type Query {
  me: User
}
```
Manager and TopManager are marked as unreachable, although they are reachable by `Query > me`.

**Expected behavior**

No error.

**Environment:**

- OS:
- `@graphql-eslint/...`: 1.1.4
- NodeJS:

**Additional context**

**Comment by tobiasdiez** [comment]
I noticed that there is now a version 2.0.1, so I tried upgrading. However, now even types inhering a used interface are marked as unreachable, e.g. as in
```
interface User {
  id: ID!
}

type Manager implements User {
  id: ID!
}

type Query {
  me: User
}

``` 
Manager is marked as unreachable.

(see https://github.com/JabRef/JabRefOnline/pull/350 for a real-life example).

**Comment by B2o5T** [comment]
Thanks for the detailed report, I'll fix this soon

**Comment by B2o5T** [comment]
Hi @tobiasdiez, can you try this alpha version `2.1.0-alpha-575822d.0` and tell me if this fixes your issues?

**Comment by tobiasdiez** [comment]
Thanks! The problem with the interfaces seems to be fixed indeed. However, all input types are now marked as unreachable.

**Comment by B2o5T** [comment]
Can you provide some code examples? This is interested because we have a test that covers `input` type

**Comment by tobiasdiez** [comment]
It concerned all input types in this file: https://github.com/JabRef/JabRefOnline/blob/main/api/documents/schema.graphql
I don't have time right now, but the coming days I will experiment and try to find a minimal example.

**Comment by B2o5T** [comment]
I don't see problems, I got this result
```sh
    2:2  warning  Type "Date" is unreachable                        @graphql-eslint/no-unreachable-types
    4:2  warning  Type "DocumentFilters" is unreachable             @graphql-eslint/no-unreachable-types
   14:2  warning  Type "User" is unreachable                        @graphql-eslint/no-unreachable-types
   27:2  warning  Type "AddPersonInput" is unreachable              @graphql-eslint/no-unreachable-types
   31:2  warning  Type "AddOrganizationInput" is unreachable        @graphql-eslint/no-unreachable-types
   41:2  warning  Type "AddEntityInput" is unreachable              @graphql-eslint/no-unreachable-types
   47:2  warning  Type "Translatable" is unreachable                @graphql-eslint/no-unreachable-types
   57:2  warning  Type "Translated" is unreachable                  @graphql-eslint/no-unreachable-types
   77:2  warning  Type "AddTranslatedInput" is unreachable          @graphql-eslint/no-unreachable-types
   82:2  warning  Type "Institution" is unreachable                 @graphql-eslint/no-unreachable-types
  124:2  warning  Type "Event" is unreachable                       @graphql-eslint/no-unreachable-types
  280:2  warning  Type "Booklike" is unreachable                    @graphql-eslint/no-unreachable-types
  301:2  warning  Type "PartOfSeries" is unreachable                @graphql-eslint/no-unreachable-types
  318:2  warning  Type "HasParent" is unreachable                   @graphql-eslint/no-unreachable-types
  347:2  warning  Type "HasBooklikeParent" is unreachable           @graphql-eslint/no-unreachable-types
  359:2  warning  Type "HasMultiVolumeParent" is unreachable        @graphql-eslint/no-unreachable-types
  401:2  warning  Type "DocumentMultiVolume" is unreachable         @graphql-eslint/no-unreachable-types
  456:2  warning  Type "AddUserDocumentInput" is unreachable        @graphql-eslint/no-unreachable-types
  466:2  warning  Type "UpdateUserDocumentInput" is unreachable     @graphql-eslint/no-unreachable-types
  486:2  warning  Type "AddJournalArticleInput" is unreachable      @graphql-eslint/no-unreachable-types
  489:2  warning  Type "AddProceedingsArticleInput" is unreachable  @graphql-eslint/no-unreachable-types
  492:2  warning  Type "AddThesisInput" is unreachable              @graphql-eslint/no-unreachable-types
  498:2  warning  Type "User" is unreachable                        @graphql-eslint/no-unreachable-types
```
`scalar Date` never reached

`input DocumentFilters` reachable by `type User` that don't reachable by some root level type like `Query`, `Mutation` or `Subscription` so types `User` and `DocumentFilters` marked as unreachables

`input AddPersonInput` reachable only by `input AddEntityInput` that reachable only by `input AddTranslatedInput` that don't reachable by some root level type so all inputs types `AddPersonInput`, `AddEntityInput` and `AddTranslatedInput` marked as unreachables

Tell me if I'm wrong somewhere

**Comment by tobiasdiez** [comment]
Thanks for looking into this. The inputs are reachable from Mutation via
https://github.com/JabRef/JabRefOnline/blob/4c2fe6bd69852342aa65de51d708d7d40f043f47/api/documents/schema.graphql#L480-L483

Maybe the issue is that my schema is defined in multiple files, that are stitched together.

**Comment by B2o5T** [comment]
Yes, I found the issue, I'll fix it soon, thanks 👍

**Comment by B2o5T** [comment]
fixed, you can try `2.1.0-alpha-e71f596.0`

**Comment by tobiasdiez** [comment]
Thanks a lot for the quick fix! It is working for me.

**Comment by dotansimha** [comment]
Fixed in `@graphql-eslint/eslint-plugin@2.1.0`. Thanks @B2o5T !
